cmd/ctr/run.go

```diff
@@ -189,29 +195,69 @@ var runCommand = cli.Command{
 		if err != nil {
 			return err
 		}
-		events, err := containers.Events(gocontext.Background(), &execution.EventsRequest{})
+		events, err := containers.Events(ctx, &execution.EventsRequest{})
 		if err != nil {
 			return err
 		}
-		abs, err := filepath.Abs(context.String("rootfs"))
+
+		provider, err := getContentProvider(context)
 		if err != nil {
 			return err
 		}
-		// for ctr right now just do a bind mount
-		rootfs := []*mount.Mount{
-			{
-				Type:   "bind",
-				Source: abs,
-				Options: []string{
-					"rw",
-					"rbind",
-				},
-			},
+
+		rootfsClient, err := getRootFSService(context)
+		if err != nil {
+			return err
+		}
+
+		db, err := getDB(context, false)
+		if err != nil {
+			return errors.Wrap(err, "failed opening database")
+		}
+		defer db.Close()
+
+		tx, err := db.Begin(false)
+		if err != nil {
+			return err
+		}
+		defer tx.Rollback()
+
+		ref := context.Args().First()
+
+		im, err := image.Get(tx, ref)
+		if err != nil {
+			return errors.Wrapf(err, "could not resolve %q", ref)
 		}
+		// let's close out our db and tx so we don't hold the lock whilst running.
+		tx.Rollback()
```

Comment:
Sorry, I misunderstood you. `tx.Rollback()` is closed early here so that we avoid holding on to the transaction for the rest of the command.

> is it safe when they get called twice?

As far as I know, yes.

`db.Close()` should probably be called earlier, but in practice, this should all happen over a GRPC interface anyways.